test/pylib/resource_gather.py

```diff
@@ -0,0 +1,166 @@
+        self.sqlite_writer = SQLiteWriter(self.db_path)
+
+    def make_cgroup(self):
+        os.makedirs(self.cgroup_path, exist_ok=True)
+
+    def put_process_to_cgroup(self, pid):
+        try:
+            with open(self.cgroup_path / 'cgroup.procs', "a") as cgroup:
+                cgroup.write(str(pid))
+        except ProcessLookupError:
+            logging.getLogger().log(
+                level=logging.WARNING,
+                msg=f"Process finished too quick. No metrics will be logged",
+            )
+
+    def get_test_metrics(self) -> TestMetricRecord:
+        test_metrics = TestMetricRecord()
+        uname_split = self.test.uname.split('.')
+        test_metrics.directory = uname_split[0]
+        test_metrics.run_id = uname_split[-1]
+        test_metrics.test_name = self.test.uname
+        test_metrics.mode = self.test.mode
+        test_metrics.architecture = platform.machine()
+        test_metrics.time_taken = round(int(self.test.time_end - self.test.time_start))
+        test_metrics.time_start = datetime.fromtimestamp(self.test.time_start).strftime(DATE_TIME_TEMPLATE)
+        test_metrics.time_end = datetime.fromtimestamp(self.test.time_end).strftime(DATE_TIME_TEMPLATE)
+        test_metrics.success = self.test.success
+        with open(self.cgroup_path / 'memory.peak', 'r') as file:
+            test_metrics.memory_peak = file.read()
+
+        if (self.cgroup_path / 'cpu.stat').exists():
+            with open(self.cgroup_path / 'cpu.stat', 'r', ) as file:
+                self._parse_cpu_stat(file, test_metrics)
+        return test_metrics
+
+    def write_metrics_to_db(self, metrics: TestMetricRecord, success: bool = False) -> None:
+        metrics.success = success
+        metrics_dict = asdict(metrics, filter=lambda attr, value: value is not None)
+        self.sqlite_writer.write_row(metrics_dict, METRICS_TABLE)
+
+    @staticmethod
+    def _parse_cpu_stat(file: TextIO, metrics: TestMetricRecord):
+        for stat in ['user_usec', 'system_usec', 'usage_usec']:
+            for line in file.readlines():
+                if line.startswith(stat):
+                    setattr(metrics, stat, float(line.split()[-1]))
+
+
+def get_resource_gather(is_switched_on, test, tmpdir: str) -> ResourceGather:
+    if is_switched_on:
+        return ResourceGatherOn(test, tmpdir)
+    else:
+        return ResourceGatherOff(test, tmpdir)
+
+
+def setup_cgroup(is_required: bool):
+    if is_required:
+        # check that we are in podman or in docker
+        is_podman = os.access("/run/.containerenv", os.F_OK)
+        if is_podman:
+            subprocess.run(['sudo', 'chown', '-R', f"{getpass.getuser()}:{getpass.getuser()}", '/sys/fs/cgroup'],
+                           check=True)
+        else:
+            subprocess.run(
+                [
+                    "sudo",
+                    "mount",
+                    "-o",
+                    "remount,rw,memory_recursiveprot,nsdelegate",
```

Comment:
Please add some comments about these options.

You're affecting the callers environment. Is this okay?

Reply:
Removed these options as redundant in our case. 
This intended to be used in containers. Without toolchain this part will break.